core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -378,7 +378,7 @@ public static Object coerceListToArray(@Nullable List<?> val, boolean homogenize
       return new String[]{null};
     } else {
       if (val != null) {
-        return val.toArray();
+        return new String[0];
```

Comment:
nit: we can cache the instance of an empty array. A similar technique can be applied for the single element array of null above. I think we can do this as a follow-up.